Classes/UT3Nightshade.uc

```diff
@@ -278,91 +306,160 @@ function CheckState()
 	ArmExtend [121]
 	ArmExtendIdle [1]
 	ArmRelease [1]
-	ArmRetract [121] (Made by me. Reverse of ArmExtend)
+	ArmRetract [121] (Made by zeluisping; Reverse of ArmExtend)
 	Idle [1]
 	*/
-	if (CurrentState == VS_Deploying && CurrentAnim == "ArmExtend" && !IsAnimating())
-	{
-		CurrentState = VS_Deployed;
+	if (CurrentState == VS_Deploying && CurrentAnim == "ArmExtend" && !IsAnimating()) {
+		ChangeDeployState(VS_Deployed);
+	} else if (CurrentState == VS_Undeploying && CurrentAnim == "ArmRetract" && !IsAnimating()) {
+		ChangeDeployState(VS_Cloaked);
 	}
-	else if (CurrentState == VS_Undeploying && CurrentAnim == "ArmRetract" && !IsAnimating())
-	{
-		if (ArmMine != None)
-			ArmMine.Destroy();
-		// Reset these.
-		MaxThrustForce = Default.MaxThrustForce;
-		MaxStrafeForce = Default.MaxStrafeForce;
-		TurnTorqueMax = Default.TurnTorqueMax;
-		Cloak(true); // KMaxSpeed gets reset here.
-		CurrentState = VS_Cloaked;
+}
+function ChangeDeployState(VState NewState) {
+	local VState PrevState;
+
+	PrevState = CurrentState;
+	CurrentState = NewState;
+
+	switch (CurrentState) {
+		case VS_Deployed:
+			// sanity check
+			if (PrevState == VS_Deploying) {
+				// Lock main weapon rotation
+				Weapons[0].YawConstraintDelta = 0;
+			//	Weapons[0].PitchConstraintDelta = 0;
+
+				// Setup camera (SpecialCalcView forces third person when deployed)
+				TPCamLookat = DeployedTPCamLookat;
+				LastDesiredTPCamDistance = DesiredTPCamDistance;
+				DesiredTPCamDistance = DeployedTPCamDistance;
+			}
+			break;
+		case VS_Undeployed:
+			// fallthrough
+		case VS_Cloaked:
+			Cloak(CurrentState == VS_Cloaked);
+
+			if (PrevState == VS_Undeploying) {
+				if (ArmMine != None) {
+					ArmMine.Destroy();
+				}
+
+				// Reset movement
+				MaxThrustForce = Default.MaxThrustForce;
+				MaxStrafeForce = Default.MaxStrafeForce;
+				TurnTorqueMax = Default.TurnTorqueMax;
+
+				// Reset main weapon rotation constraint
+				Weapons[0].YawConstraintDelta = Weapons[0].default.YawEndConstraint - Weapons[0].default.YawStartConstraint;
+			//	Weapons[0].PitchConstraintDelta = Weapons[0].default.PitchEndConstraint - Weapons[0].default.PitchStartConstraint;
+
+				// Reset third person camera
+				TPCamLookat = default.TPCamLookat;
+				DesiredTPCamDistance = LastDesiredTPCamDistance;
+			}
+			break;
+		case VS_Deploying:
+			if (PrevState == VS_Cloaked) {
+				Cloak(false);
+			}
+
+			PlaySound(DeploySnd, SLOT_None);
+			PlayAnim('ArmExtend', 1.6, 0.2);
+			CurrentAnim = "ArmExtend";
+
+			// Do not allow to move or rotate.
+			MaxThrustForce = 0.0;
+			MaxStrafeForce = 0.0;
+			KarmaParamsRBFull(KParams).KMaxSpeed = 0.0;
+			TurnTorqueMax = 0.0;
+			break;
+		case VS_Undeploying:
+			SetBoneRotation('ArmShoulder', rot(0,0,0), 0, 1);
+			PlaySound(UndeploySnd, SLOT_None);
+			PlayAnim('ArmRetract', 4.8, 0.2);
+			CurrentAnim = "ArmRetract";
+			break;
+		default:
+			// crash
+			break;

```

Comment:
There are no other states, so this will never be run. Instead, you should make `VS_Undeployed` and `VS_Cloaked` the default, that will save some lines of code.

Reply:
Will do!